staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/union_content.go

```diff
@@ -55,7 +55,12 @@ func (c unionCAContent) CurrentCABundleContent() []byte {
 }
 
 // CurrentCABundleContent provides ca bundle byte content
-func (c unionCAContent) VerifyOptions() x509.VerifyOptions {
+func (c unionCAContent) VerifyOptions() (x509.VerifyOptions, bool) {
+	currCABundle := c.CurrentCABundleContent()
+	if len(currCABundle) == 0 {
```

Comment:
does this mean it is not possible to express "use system roots"?

Reply:
> does this mean it is not possible to express "use system roots"?

I'm pretty sure that we explicitly chose to have that *not* be the meaning of empty for client certificate validation, since that would imply that any officially signed client cert in the world is a valid identity for a given cluster instead of selectively choosing.  On the server verification side, it's very important.  On the client verification side, I question its importance and whether we ever supported it.

Reply:
I think you're right